generator/format/mo/option.go

```diff
@@ -2,42 +2,44 @@ package mo
 
 import (
 	"fmt"
+	"go/types"
+
 	"github.com/11wizards/go-to-dart/generator/format"
-	"go/ast"
 )
 
 type OptionFormatter struct {
 	format.TypeFormatterBase
 }
 
-func (f *OptionFormatter) under(expr ast.Expr) (format.TypeFormatter, ast.Expr) {
-	e := expr.(*ast.IndexExpr).Index
+func (f *OptionFormatter) under(expr types.Type) (format.TypeFormatter, types.Type) {
+	e := expr.(*types.Named).TypeArgs().At(0)
 	formatter := f.Registry.GetTypeFormatter(e)
 	return formatter, e
 }
 
-func (f *OptionFormatter) CanFormat(expr ast.Expr) bool {
-	if x, ok := expr.(*ast.IndexExpr); ok {
-		if y, ok := x.X.(*ast.SelectorExpr); ok && y.X.(*ast.Ident).Name == "mo" && y.Sel.Name == "Option" {
+func (f *OptionFormatter) CanFormat(expr types.Type) bool {
+	if namedType, ok := expr.(*types.Named); ok {
+		if namedType.Obj().Type().String() == "github.com/samber/mo.Option[T any]" {
```

Comment:
How will it work if T will be specified as something else than `any`?

Reply:
That could only happen if mo.Option[T any] gets refactored to be something else in the original package -- I'd assume this would be a major breaking change in samber/mo! I think this is pretty safe for now.